File: algorithm/501. Find Mode in Binary Search Tree/501.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> findMode(TreeNode* root) {
        map<int,int> m;
        int tem=helper(root,m);
        vector<int> result;
        for(auto i:m)
        {
            if(i.second==tem)
                result.push_back(i.first);
        }
        return result;
    }
    int helper(TreeNode* root, map<int,int> & m )
    {
        if(root)
        {
            m[root->val]++;
        }
        else
            return 0;
        return max(m[root->val],max(helper(root->left,m),helper(root->right,m)));
    }
    
};

File: algorithm/501. Find Mode in Binary Search Tree/501.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def findMode(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        def helper(root, map):
            if root:
                m[root.val]=m.get(root.val,0)+1
                helper(root.left,m)
                helper(root.right,m)
        m={}
        helper(root,m)
        if len(m)==0:
            return []
        k=max(m.values())
        return [i for [i,j] in m.items() if j==k]
                
